lib/create-space-api.js

```diff
@@ -513,6 +514,81 @@ export default function createSpaceApi ({
   }
 
   /**
+   * Creates a Asset based on files. After creation, call asset.processForLocale or asset.processForAllLocales to start asset processing.
+   * @memberof ContentfulSpaceAPI
+   * @see {Asset.Asset}
+   * @param {object} data - Object representation of the Asset to be created. Note that the field object should have an uploadFrom property on asset creation, which will be removed and replaced with an url property when processing is finished.
+   * @param {object} data.fields.file.[LOCALE].file - Can be a string, an ArrayBuffer or a Stream.
+   * @return {Promise<Asset.Asset>} Promise for the newly created Asset
+   * @example
+   * space.createAssetFromFiles({
+   *   fields: {
+   *     file: {
+   *       'en-US': {
+   *          contentType: 'image/jpeg',
+   *          fileName: 'filename_english.jpg',
+   *          file: readFileSync('path/to/filename_english.jpg')
+   *       },
+   *       'de-DE': {
+   *          contentType: 'image/svg',
+   *          fileName: 'filename_german.svg',
+   *          file: '<svg><path fill="red" d="M50 50h150v50H50z"/></svg>'
+   *       }
+   *     }
+   *   }
+   * })
+   * .then(e => console.log(e))
+   */
+  function createAssetFromFiles (data) {
+    const { file } = data.fields
+    return Promise.all(
+      Object.keys(file).map((locale) => createUpload(locale, file[locale]))
+    )
+    .then((uploads) => {
+      data.fields.file = uploads.reduce((fieldsData, upload) => {
+        return Object.assign(fieldsData, upload)
+      }, {})
+      return createAsset(data)
+    })
+  }
+
+  /**
+   * Creates a Upload.
+   * @memberof ContentfulSpaceAPI
+   * @param {object} locale - The locale of the file
+   * @param {object} data - Can be a string, an ArrayBuffer or a Stream.
+   * @return {Promise<Object>} File field value for the given locale.
+   * @example
+   * const fileToUpload = readFileSync('path/to/filename_english.jpg')
+   * space.createUpload('en-US', fileToUpload)
+   */
+  function createUpload (locale, data) {
+    const { file, fileName, contentType } = data
+    httpUpload.post('uploads', file, {
+      headers: {
+        'Content-Type': 'application/octet-stream'
+      }
+    })
+    .then((uploadResponse) => {
+      const { id } = uploadResponse.data.sys
+
+      return {
```

Comment:
I think you are making so many changes to the original response, why not just returning the response object, because a user will expect to have an upload object and not an asset data.